prow/config/config.go

```diff
@@ -261,18 +261,24 @@ type Branding struct {
 }
 
 // Load loads and parses the config at path.
-func Load(prowConfig, jobConfig string) (*Config, error) {
-	nc, err := loadConfig(prowConfig, jobConfig)
+func Load(prowConfig, jobConfig string) (c *Config, err error) {
+	// we never want config loading to take down the prow components
+	defer func() {
+		if r := recover(); r != nil {
```

Comment:
`recover` is one of the corners of go I've not really had a use case for before, I'm not positive if this is what we want. 
It's used very sparingly in the stdlib, the only well known and accepted example I know of is `encoding/json` for marshal/unmarshal, so this does seem appropriate but i'm not sure if this is exactly how we want to do it..

Reply:
`Load` is typically called from a background goroutine. If a panic happens on the main thread will it be caught by this recover statement?

Reply:
defer + recover handle the current goroutine. https://golang.org/pkg/builtin/#recover

> The recover built-in function allows a program to manage behavior of a panicking goroutine. Executing a call to recover inside a deferred function (but not any function called by it) stops the panicking sequence by restoring normal execution and retrieves the error value passed to the call of panic. If recover is called outside the deferred function it will not stop a panicking sequence. In this case, or when the goroutine is not panicking, or if the argument supplied to panic was nil, recover returns nil. Thus the return value from recover reports whether the goroutine is panicking.

Whatever goroutine calls `Load()` should be recovered.
See also: https://blog.golang.org/defer-panic-and-recover

> Recover is a built-in function that regains control of a panicking goroutine. Recover is only useful inside deferred functions. During normal execution, a call to recover will return nil and have no other effect. If the current goroutine is panicking, a call to recover will capture the value given to panic and resume normal execution.